### PR TITLE
bug-fix: no need to check gtid of Mariadb

### LINKED ISSUE #1319: When Mariadb enable-gtid=true, executing operation-source create failed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

When Mariadb enable-gtid=true, executing operation-source create failed
image

2. What did you expect to see?

3. What did you see instead?

```
[tidb@node5172 conf]$ tiup dmctl  --master-addr 172.16.5.172:8361 operate-source create ./mariadb_source.yaml 
Starting component `dmctl`: /home/tidb/.tiup/components/dmctl/v2.0.0/dmctl/dmctl --master-addr 172.16.5.172:8361 operate-source create ./mariadb_source.yaml
{
    "result": false,
    "msg": "[code=10001:class=database:scope=not-set:level=high], Message: database driver error, RawCause: sql: no rows in result set, Workaround: Please check the database connection and the database config in configuration file.",
    "sources": [
    ]
}

[tidb@node5172 conf]$ cat ./mariadb_source.yaml 
source-id: "mysql-replica-02"
enable-gtid: true

from:
  host: "172.16.5.127"
  port: 3306
  user: "root"
  password: "password"
```

4. Versions of the cluster
```
[tidb@node5172 conf]$ tiup dm display dm1               
Starting component `dm`: /home/tidb/.tiup/components/dm/v1.2.5/tiup-dm display dm1
Cluster type:    dm
Cluster name:    dm1
Cluster version: v2.0.0
SSH type:        builtin
ID                 Role        Host          Ports      OS/Arch       Status     Data Dir                           Deploy Dir
--                 ----        ----          -----      -------       ------     --------                           ----------
172.16.5.172:8361  dm-master   172.16.5.172  8361/8391  linux/x86_64  Healthy|L  /tidb/dmwang-data/dm-master-8361   /tidb/dmwang-deploy/dm-master-8361
172.16.4.107:8262  dm-worker   172.16.4.107  8262       linux/x86_64  Free       /tidb/dmwang-data/dm-worker-8262   /tidb/dmwang-deploy/dm-worker-8262
172.16.4.107:3330  grafana     172.16.4.107  3330       linux/x86_64  Up         -                                  /tidb/dmwang-deploy/grafana-3330
172.16.4.107:9390  prometheus  172.16.4.107  9390       linux/x86_64  Up         /tidb/dmwang-data/prometheus-9390  /tidb/dmwang-deploy/prometheus-9390
Total nodes: 4
```

- Upstream MySQL/MariaDB server version:

```
root@172.16.5.127 :(none)>select version();
+-----------------+
| version()       |
+-----------------+
| 10.1.48-MariaDB |
+-----------------+
1 row in set (0.00 sec)
 ```

5. current status of DM cluster (execute `query-status` in dmctl)

6. Operation logs
   - Please upload `dm-worker.log` for every DM-worker instance if possible
   - Please upload `dm-master.log` if possible
   - Other interesting logs
   - Output of dmctl's commands with problems
   
7. Configuration of the cluster and the task
   - `dm-worker.toml` for every DM-worker instance if possible
   - `dm-master.toml` for DM-master if possible
   - task config, like `task.yaml` if possible
   - `inventory.ini` if deployed by DM-Ansible

8. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus for DM if possible

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #1319 

### What is changed and how it works?
- fix gitd check of Mariadb
- better incompatible ddl error message
- test for auto sql_mode
TODO:
  add mariadb test for integration test or chaos test
### Check List 

Tests 

 - Integration test
 - Manual test (add detailed scripts or steps below)
    deploy mariadb locally and run integration test

### CODE REVIEW

File: dm/config/source_config.go [hunk 1]
```diff
@@ -232,7 +232,8 @@ func (c *SourceConfig) Adjust(ctx context.Context, db *sql.DB) (err error) {
 		}
 	}
 
-	if c.EnableGTID {
+	// MariaDB automatically enabled gtid after 10.0.2, refer to https://mariadb.com/kb/en/gtid/#using-global-transaction-ids
+	if c.EnableGTID && c.Flavor == mysql.MySQLFlavor {
```
Comment (csuzhangxc) [inline dm/config/source_config.go:236]:
```suggestion
	if c.EnableGTID && c.Flavor != mysql.MariaDBFlavor {
```

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

**Review by csuzhangxc — COMMENTED** [review]
LGTM

File: tests/sharding/run.sh [hunk 2]
```diff
@@ -17,9 +17,8 @@ EOF
 }
 
 function run() {
-    run_sql "SET @@GLOBAL.SQL_MODE='NO_ZERO_IN_DATE,NO_ZERO_DATE'" $MYSQL_PORT1 $MYSQL_PASSWORD1
```
Comment (lance6716) [inline tests/sharding/run.sh:20]: in `db1.increment.sql` there's some zero datetime, so maybe this is a case to ensure without `STRICT_TRANS_TABLES` `NO_ZERO_DATE` doesn't block?

File: dm/config/source_config.go [hunk 3]
```diff
@@ -232,7 +232,8 @@ func (c *SourceConfig) Adjust(ctx context.Context, db *sql.DB) (err error) {
 		}
 	}
 
-	if c.EnableGTID {
+	// MariaDB automatically enabled gtid after 10.0.2, refer to https://mariadb.com/kb/en/gtid/#using-global-transaction-ids
```
Comment (lance6716) [inline dm/config/source_config.go:235]: I guess there's no way to disable master's GTID right 🤔

**Review by lance6716 — COMMENTED** [review]
rest LGTM, waiting comments to be addressed

File: tests/sharding/run.sh (hunk 2 above)
Comment (GMHDBJD) [reply tests/sharding/run.sh:20]: Add it back for source1. Seems mariadb not allow zero_date even without `strict_trans_tables`

File: dm/config/source_config.go (hunk 3 above)
Comment (GMHDBJD) [reply dm/config/source_config.go:235]: Yes, I can't find a way to disable it.

**Review by lance6716 — APPROVED** [review]
LGTM

**Pull request merged**